py3dtilers/CityTiler/citym_cityobject.py

```diff
@@ -10,24 +10,25 @@ class CityMCityObject(ObjectToTile):
     _CityObject. (cf 3DCityDB Version 3.3.0 Documentation).
     """
 
-    def __init__(self, database_id=None):
+    def __init__(self, database_id=None, gml_id=None):
         super().__init__(database_id)
+        self.set_gml_id(gml_id)
 
     def get_database_id(self):
         return super().get_id()
 
     def set_database_id(self, id):
-        return super().set_id(id)
+        super().set_id(id)
 
     def set_gml_id(self, gml_id):
-        self.gml_id = gml_id
+        super().set_data(gml_id)
 
     def get_gml_id(self):
         """
         :return: the (city)gml identifier of an object that should be encountered
                 in the database.
         """
-        return self.gml_id
+        return super().get_data()
```

Comment:
Correct me if I am wrong, the function name is not consistent with what is being returned by this new change. I do not understand the need for this change.

Reply:
The change is more evident when you look at the changes to the abstract class in py3dtilers/Common/object_to_tile.py

Here all objects to tile have a `self.data` parameter which can store information which will later be stored in the batch table. This change also comes with getters and setters that can be used to access and modify `self.data`.

In CityTiler's case this parameter can be used to store gml:ids, but other tilers, for example IFCTiler, could use `self.data` to store other information about an object to tile such as ids or height.